lib/services/product-manager.js

```diff
@@ -384,15 +416,8 @@ export default class ProductManager {
   }
 
   async changeProductType (product, newProductTypeId) {
-    if (product.masterData.published)
-      await this._unpublishProduct(product)
-
-    // fetch from product projection because it has same structure
-    // as product draft. It's easier for product creation later in the process
-    const { body: productProjection } = await this.client.productProjections
-      .staged(true)
-      .byId(product.id)
-      .fetch()
+    this.logger.debug('Changing productType', product.id)
+    const productProjection = await this.getProductProjectionById(product.id)
```

Comment:
As you created `transformProductToProjection`, do you think we can use it here? It will save one request.

Reply:
Updated here a152ebb .. hopefully, the `transformProductToProjection` method does not lose any fields